internal/rbd/rbd_util.go

```diff
@@ -1366,15 +1366,6 @@ func (rv *rbdVolume) cloneRbdImageFromSnapshot(
 		}
 	}()
 
-	if pSnapOpts.isEncrypted() {
```

Comment:
This is removed and moved to a higher level
Since `cloneRbdImageFromSnapshot` will be called for intermediate temp clone creation too which will not have encryption initialized.

Reply:
better place this in a new helper function, instead of having multiple (very) similar copies

Reply:
> better place this in a new helper function, instead of having multiple (very) similar copies

@nixpanic does it sound better to move this check within `.copyEncryptionConfig()`?
that way copy is done only when parent image is encrypted.

Reply:
I guess so, but you'll need to make that clear in the comment documenting the function

Reply:
> I guess so, but you'll need to make that clear in the comment documenting the function

@nixpanic 
Thats done already too https://github.com/ceph/ceph-csi/pull/3066/files#diff-8133ae76a4b0e57aab17837008b76f106b07d4c1cc8631762106e3a79a899247
PTAL